invenio_communities/assets/semantic-ui/js/invenio_communities/members/components/SuccessIcon.js

```diff
@@ -17,23 +20,29 @@ export class SuccessIcon extends Component {
         clearTimeout(this.successTimer);
       }
 
-      this.successTimer = setTimeout(onTimeOut, timeOutDelay);
+      this.successTimer = setTimeout(this.handleOnTimeOut, timeOutDelay);
     }
   }
 
   componentWillUnmount() {
     this.successTimer && clearTimeout(this.successTimer);
   }
 
-  render() {
-    const { success } = this.props;
+  handleOnTimeOut = () => {
+    this.setState({ success: false });
+  };
 
-    return success ? <Icon color="green" name="checkmark" /> : null;
+  render() {
+    const { success } = this.state;
+    return success && <Icon color="green" name="checkmark" />;
```

Comment:
@NRodriguezcuellar self contained state management - this is what I mentioned before about single component responsibility